python/lance/__init__.py

```diff
@@ -13,3 +33,18 @@ def dataset(uri: str):
     """
     fmt = LanceFileFormat()
     return ds.dataset(uri, format=fmt)
+
+
+def write_table(table: pa.Table, destination: Union[str, Path], primary_key: str):
```

Comment:
maybe add a convenience to auto generate a pk column?

Reply:
should we push that to the application / db level?

Reply:
If we want people to use it as a python library then it's probably a good idea to have it. Could be in a wrapper function or something? Should also check for uniqueness there as well.